src/shogun2-core/src/main/java/de/terrestris/shogun2/model/Plugin.java

```diff
@@ -0,0 +1,205 @@
+
+/**
+ *
+ */
+@Entity
+@Table
+@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
+public class Plugin extends PersistentObject {
+
+	/**
+	 *
+	 */
+	private static final long serialVersionUID = 1L;
+
+	/** A name (or display name) */
+	private String name;
+
+	/** the class name of the plugin **/
+	@Column(unique = true)
+	private String className;
+
+	/** the xtype of the plugin **/
+	@Column(unique = true)
+	private String xtype;
+
+	/** the JavaScript (JS) code of the plugin **/
+	@Column(length = Integer.MAX_VALUE)
+	private String sourceCode;
+
+	/** the Cascading Style Sheets (CSS) of the plugin **/
+	@Column(length = Integer.MAX_VALUE)
+	private String styleSheet;
+
+	/** A list of assigned {@link PluginUploadFile}s. */
+	@ManyToMany
+	@JoinTable(
+		joinColumns = { @JoinColumn(name = "PLUGIN_ID") },
+		inverseJoinColumns = { @JoinColumn(name = "FILE_ID") }
+	)
+	@JsonIgnore
+	private Set<File> fileUploads = new HashSet<File>();
+
+	/**
+	 * Constructor
+	 */
+	public Plugin() {
+	}
+
+	/**
+	 * @return the name
+	 */
+	public String getName() {
+		return name;
+	}
+
+	/**
+	 * @param name the name to set
+	 */
+	public void setName(String name) {
+		this.name = name;
+	}
+
+	/**
+	 * @return the className
+	 */
+	@Column(unique = true, nullable = false)
+	public String getClassName() {
+		return className;
+	}
+
+	/**
+	 * @param className the className to set
+	 */
+	public void setClassName(String className) {
+		this.className = className;
+	}
+
+	/**
+	 * @return the xtype
+	 */
+	@Column(unique = true, nullable = false)
+	public String getXtype() {
+		return xtype;
+	}
+
+	/**
+	 * @param xtype the xtype to set
+	 */
+	public void setXtype(String xtype) {
+		this.xtype = xtype;
+	}
+
+	/**
+	 * @return the sourceCode
+	 */
+	public String getSourceCode() {
+		return sourceCode;
+	}
+
+	/**
+	 * @param sourceCode the sourceCode to set
+	 */
+	public void setSourceCode(String sourceCode) {
+		this.sourceCode = sourceCode;
+	}
+
+	/**
+	 * @return the styleSheet
+	 */
+	public String getStyleSheet() {
+		return styleSheet;
+	}
+
+	/**
+	 * @param styleSheet the styleSheet to set
+	 */
+	public void setStyleSheet(String styleSheet) {
+		this.styleSheet = styleSheet;
+	}
+
+	/**
+	 * @return the fileUploads
+	 */
+	public Set<File> getFileUploads() {
+		return fileUploads;
+	}
+
+	/**
+	 * @param fileUploads the fileUploads to set
+	 */
+	public void setFileUploads(Set<File> fileUploads) {
+		this.fileUploads = fileUploads;
+	}
+
+	/**
+	 * @see java.lang.Object#hashCode()
+	 *
+	 * According to
+	 * http://stackoverflow.com/questions/27581/overriding-equals-and-hashcode-in-java
+	 * it is recommended only to use getter-methods when using ORM like Hibernate
+	 */
+	@Override
+	public int hashCode() {
+		return new HashCodeBuilder(47, 19) // two randomly chosen prime numbers
+			.appendSuper(super.hashCode())
+			.append(getName())
+			.append(getClassName())
+			.append(getXtype())
```

Comment:
Nice catch (regarding `sourceCode` and `styleSheet`). I'll add them now, but leave out `fileUploads` to avoid lazy loading of related entities...